libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginOptionsActivity.kt

```diff
@@ -0,0 +1,320 @@
+                unfocusedIndicatorColor = colorScheme.hintTextColor,
+                unfocusedLabelColor = colorScheme.hintTextColor,
+                unfocusedContainerColor = Color.Transparent,
+                unfocusedTextColor = colorScheme.onSecondary,
+                cursorColor = colorScheme.tertiary,
+            ),
+        )
+    }
+}
+
+@Composable
+fun BootConfigItem(name: String, value: String?) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(start = PADDING_SIZE.dp, end = PADDING_SIZE.dp, top = PADDING_SIZE.dp)
+    ) {
+        Text(
+            text = name,
+            fontSize = TEXT_SIZE.sp,
+            color = colorScheme.onSecondary,
+        )
+        Text(
+            text = value ?: "",
+            fontSize = TEXT_SIZE.sp,
+            color = colorScheme.onSecondaryContainer,
+            modifier = Modifier.padding(top = PADDING_SIZE.dp),
+        )
+    }
+}
+
+@Composable
+fun LoginOptionsScreen(
+    innerPadding: PaddingValues,
+    useWebServer: MutableLiveData<Boolean>,
+    useHybridToken: MutableLiveData<Boolean>,
+    supportWelcomeDiscovery: MutableLiveData<Boolean>,
+    bootConfig: BootConfig = BootConfig.getBootConfig(LocalContext.current),
+) {
+    var useDynamicConfig by remember { mutableStateOf(false) }
+
+    Column(
+        modifier = Modifier
+            .padding(innerPadding)
+            .fillMaxSize()
+            .verticalScroll(rememberScrollState()),
+    ) {
+        OptionToggle("Use Web Server Flow", useWebServer)
+        OptionToggle("Use Hybrid Auth Token", useHybridToken)
+        OptionToggle("Support Welcome Discovery", supportWelcomeDiscovery)
+
+        HorizontalDivider()
+
+        Text(
+            text = "Boot Config File",
+            fontWeight = FontWeight.Bold,
+            modifier = Modifier.padding(PADDING_SIZE.dp)
+        )
+
+        val scopes = bootConfig.oauthScopes ?: emptyArray<String>()
+        val bootConfigList = listOf(
+            "Consumer Key" to bootConfig.remoteAccessConsumerKey,
+            "Redirect URI" to bootConfig.oauthRedirectURI,
+            "Scopes" to scopes.joinToString(separator = ", "),
+        )
+        bootConfigList.forEach { (name, value) ->
+            BootConfigItem(name, value)
+        }
+
+        HorizontalDivider()
+
+        Row(
+            modifier = Modifier.fillMaxWidth().padding(PADDING_SIZE.dp),
+            horizontalArrangement = Arrangement.SpaceBetween,
+        ) {
+            Text(
+                text = "Use Dynamic Boot Config",
+                modifier = Modifier.height( 50.dp).wrapContentHeight(align = Alignment.CenterVertically),
+            )
+            Switch(
+                checked = useDynamicConfig,
+                onCheckedChange = { useDynamicConfig = it },
+            )
+        }
+
+        if (useDynamicConfig) {
+            BootConfigView()
+        }
+    }
+}
+
+val previewBootConfig = object : BootConfig() {
```

Comment:
Maybe for the preview, we use a bogus remote access consumer key / redirect uri?

Reply:
Is [this bootconfig file](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/SalesforceSDK/res/values/bootconfig.xml) sticking around?